x-pack/plugins/stack_connectors/server/connector_types/stack/torq/index.ts

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { curry } from 'lodash';
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { map, getOrElse } from 'fp-ts/lib/Option';
+import { Logger } from '@kbn/core/server';
+import { ActionType, ActionTypeExecutorOptions } from '@kbn/actions-plugin/server';
+import {
+  AlertingConnectorFeatureId,
+  UptimeConnectorFeatureId,
+  SecurityConnectorFeatureId,
+  ActionTypeExecutorResult,
+} from '@kbn/actions-plugin/common';
+import { renderMustacheString } from '@kbn/actions-plugin/server/lib/mustache_renderer';
+import { request } from '@kbn/actions-plugin/server/lib/axios_utils';
+import { ValidatorServices } from '@kbn/actions-plugin/server/types';
+import { getRetryAfterIntervalFromHeaders } from '../../lib/http_response_retry_header';
+import { promiseResult, isOk, Result } from '../../lib/result_type';
+
+export type TorqActionType = ActionType<
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  ActionParamsType,
+  unknown
+>;
+export type TorqActionTypeExecutorOptions = ActionTypeExecutorOptions<
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  ActionParamsType
+>;
+
+const configSchemaProps = {
+  webhookIntegrationUrl: schema.string(),
+};
+const ConfigSchema = schema.object(configSchemaProps);
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+// secrets definition
+export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+const secretSchemaProps = {
+  token: schema.nullable(schema.string()),
+};
+const SecretsSchema = schema.object(secretSchemaProps, {
+  validate: (secrets) => {
+    if (!secrets.token) {
+      return i18n.translate('xpack.stackConnectors.torq.secrets.tokenRequiredErrorMessage', {
+        defaultMessage: 'token is required',
+      });
+    }
+  },
+});
+
+// params definition
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+const ParamsSchema = schema.object({
+  body: schema.maybe(schema.string()),
```

Comment:
It appears `body` will have to be a parseable JSON string.  I don't think there's a need for the `schema.maybe()` wrapper here, if it's always expected there will be a body.